### ISSUE TITLE
List items don't announce position in set

### ISSUE THREAD

**Issue by mitchellholly** [microsoft/fluentui#14608]
### Environment Information

- **Package version(s)**: (fill this out)
- **Browser and OS versions**: I am reproing in Edge with Narrator, on Windows 10

### Describe the issue:

I am using a List component, and when you navigate between items in the list, it does not tell you which item you are on, like "item 2 of 40". If you use CAPS + right arrow, it does announce. But if you just use regular arrow keys, in scan or off, it does not announce.

Note there was a previous issue opened about this, #10939. There wasn't a clear resolution there. Another item #11819 was opened as a follow up, and the issue there was fixed, but it pertains specifically to Details list. This issue still happens in List.

### Please provide a reproduction of the issue in a codepen:

You can see this issue on the Fluent demo page here: https://developer.microsoft.com/en-us/fluentui#/controls/web/list with the first example, "List of 5000 grid items".

#### Actual behavior:

Position is not announced. Just the text on the item is read out.

#### Expected behavior:

Expect position to be announced, "item 2 out of 40".

**Comment by paulgildea** [comment]
@mitchellholly Thanks for the added context. Super helpful.

**Comment by cschlechty** [comment]
I believe this is the design of the screen reader. These counts are giving extra context to the user in a specific mode, forcing these counts into another mode via hackery is counterproductive. If the controls are using semantic markup, the screenreader will interpret it appropriately. This issue is trying to change the screenreader behavior.
